<!DOCTYPE html>
<html>
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge,chrome=1">
<meta name="viewport" content="width=device-width, initial-scale=1.0,user-scalable=no">
<title>Clustering Tools</title>   
{% load static %}
<link href="{% static "/images/clustering.css" %}" type="text/css" rel="stylesheet">	
</head>

<body>
	
	<div class="container">
	<header> 
		<a href=""><span class="logo">Single<br/>Cell RNA Sequencing Data</span></a>
	    <nav>
	      <ul>
	        <li><a href="/upload/index/">Home</a></li>
            <li><a href="/upload/datasets/">Datasets</a></li>
	        <li class="cur"><a href="/upload/clustering/">Clustering Tools</a></li>
			<li><a href="/upload/visulization/">Integrated methods</a></li>
			<li><a href="/upload/help/">Help</a></li>
			  
          </ul>
        </nav>
	</header>
	
	<div class="left-block">

	<section class="select">
	  <form action="/upload/upload/" method="post" enctype="multipart/form-data">
		    	<h2>Select your dataset：</h2>
		  	<div class="srcin">
				<li class="lif">Dataset: &nbsp;&nbsp;&nbsp;<input type="file" name="myfile" class="file"></li>
		  		<li class="lik">K: &nbsp;<input type="text" class="k" name="kvalue"></li>
				<h5><li>Data format : samples × features</li>
                    <li>K : the number of clusters</li> </h5>
		    </div>
			  <h2 class="checktitle">Choose your algorithm：</h2>
		  <div class="block1 ac-checkbox ac-custom ac-checkmark" autocomplete="off">
			 <h3>Distance-based methods:</h3>
			  <li><input id="cb1" name="cb1" type="checkbox"><label for="cb1" data-mtpis="An algorithm for rare cell type identification in complex populations of single cells. It contains four steps: (1) Data preparation; (2) k-means clustering; (3) Identification of outlier cells. (4) Inference of final clusters.">RaceID</label></li>
			  <li><input id="cb2" name="cb2" type="checkbox"><label for="cb2" data-mtpis="An agglomerative clustering method that can generate a cell state hierarchy where each cluster branch is associated with a principal component of variation that can be used to differentiate two cell states.">pcaReduce</label></li>
			  <li><input id="cb3" name="cb3" type="checkbox"><label for="cb3" data-mtpis="A method that uses a novel statistical approach, based on document analysis techniques, to produce tree structures outlining the hierarchical relationship between single-cell samples, while identifying latent groups of genes that can provide biological insights.">cellTree</label></li>
			  <li><input id="cb4" name="cb4" type="checkbox"><label for="cb4" data-mtpis="(Clustering through Imputation and Dimensionality Reduction), an ultrafast algorithm that uses a novel yet very simple implicit imputation approach to alleviate the impact of dropouts in scRNA-seq data in a principled manner.">CIDR</label></li>
			  <li><input id="cb5" name="cb5" type="checkbox"><label for="cb5" data-mtpis="A computational pipeline for SINgle CEll RNA-seq profiling Analysis. The pipeline supports the analysis for: 1) the distinction and identification of major cell types; 2) the identification of cell type specific gene signatures; and 3) the determination of driving forces of given cell types.">SINCERA</label></li>
			  <li><input id="cb6" name="cb6" type="checkbox"><label for="cb6" data-mtpis="(single-cell consensus clustering), a user-friendly tool for unsupervised clustering, which achieves high accuracy and robustness by combining multiple clustering solutions through a consensus approach.">SC3</label></li>
			<h3>Density-based methods:</h3>
			  <li><input id="cb7" name="cb7" type="checkbox"><label for="cb7" data-mtpis="An algorithm for rare cell type detection. The most important feature in GiniClust is a novel gene selection method that is particularly suitable for rare cell type identification, borrowing ideas from the social science domain.">GiniClust</label></li>
			<h3>Graph-based methods:</h3>
			  <li><input id="cb8" name="cb8" type="checkbox"><label for="cb8" data-mtpis="(shared nearest neighbor -Cliq), a novel algorithm utilizes the concept of shared nearest neighbor that shows advantages in handling high-dimensional data. (1) Construct an SNN graph; (2) Identify clusters in the SNN graph.">SNN-Cliq</label></li>
			  <li><input id="cb9" name="cb9" type="checkbox"><label for="cb9" data-mtpis="A computational strategy to infer cellular localization by integrating single-cell RNA-seq data with in situ RNA patterns.">Seurat</label></li>
		  </div><div class="block2 ac-checkbox ac-custom ac-checkmark">
		    <h3>Matrix-based methods:</h3>
			  <li><input id="cb10" name="cb10" type="checkbox"><label for="cb10" data-mtpis="A method adapts Nonnegative Matrix Factorization to study the problem of identifying subpopulations in single-cell transcriptome data. In contrast to the conventional gene-centered view of NMF, identifying metagenes, here NMF in a cell-centered direction, identifying cell sub- types (‘metacells’).">NMF</label></li>
			  <li><input id="cb11" name="cb11" type="checkbox"><label for="cb11" data-mtpis="A divisive biclustering method based on sorting points into neighborhoods (SPIN).">BackSPIN</label></li>
		    <h3>Model-based methods:</h3>
			  <li><input id="cb12" name="cb12" type="checkbox"><label for="cb12" data-mtpis="An iterative normalization and clustering method for single-cell gene expression data.  It is formulated as a hierarchical Bayesian mixture model with cell-specific scaling that aid the iterative normalization and clustering of cells, teasing apart technical variation from biological signals.">BISCUIT</label></li>
			  <li><input id="cb13" name="cb13" type="checkbox"><label for="cb13" data-mtpis="A Dirichlet Mixture Model for clustering droplet-based Single Cell transcriptomic data. This approach explicitly models UMI count data from scRNA-Seq experi- ments and characterizes variations across different cell clusters via a Dirichlet mixture prior.">DIMM-SC</label></li>
		    <h3>Deep learning-based methods:</h3>
			  <li><input id="cb14" name="cb14" type="checkbox"><label for="cb14" data-mtpis=" (Single-cell Variational Inference), a ready-to-use scalable framework for the probabilistic representation and analysis of gene expression in single cells. It uses stochastic optimization and deep neural networks to aggregate information across similar cells and genes and to approximate the distributions that underlie observed expression values, while accounting for batch effects and limited sensitivity. It can be used for a range of fundamental analysis tasks including batch correction, visualization, clustering and differential expression.">scVI</label></li>
		  	  <li><input id="cb15" name="cb15" type="checkbox"><label for="cb15" data-mtpis=" (Deep count autoencoder), a method takes the count distribution, over-dispersion and sparsity of the scRNA-seq data into account using a negative binomial noise model with or without zero-inflation, and nonlinear gene-gene dependencies are captured.">DCA</label></li>
		  </div>
		  <div class="mailblock" id="f3">
		Mailbox：
		<input type="text" name="mail" class="ml">
		</div>
		  
		<button class="uploads" id="subbut">Run</button>
		<div id="msg"></div>
		</form>
	</section>
	</div>
		
	<div class="instr">
		<t3>Classification of scRNA-seq data clustering methods :</t3>
		<img src="{% static "images/about.jpg" %}">
		<p>Distance-based methods use distance-based clustering algorithms such as k-means and hierarchical clustering. Density-based methods employ the density-based clustering mechanism such as the DBSCAN algorithm and its variants. Graph-based methods first transform the data to a graph, over which a graph clustering algorithm is applied. The probability model-based methods cluster data based on a certain probability distribution or process. The matrix-based methods first transform the data to a matrix, and perform matrix split-ting or decomposition to cluster the data. The Deep learning-based methods learn the latent structure of features based on Denoising Auto-encoders or Variational Autoencoders, then clustering data on the latent space.
		</p>
		</div>
	</div>
	
	<script src="http://www.jq22.com/jquery/jquery-1.10.2.js"></script>
	<script src="{% static "js/mTips.js" %}" type="text/javascript" charset="utf-8"></script>
	<script type="text/javascript" src="{% static "js/svgcheckbx.js" %}"></script>
</body></html>
